lib/Doctrine/DBAL/Logging/PsrAdapter.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Doctrine\DBAL\Logging;
+
+use Psr\Log\LoggerInterface;
+
+/**
+ * Logs every query as a debug message
+ */
+final class PsrAdapter implements SQLLogger
+{
+    /** @var LoggerInterface */
+    private $logger;
+
+    public function __construct(LoggerInterface $logger)
```

Comment:
It’s not a hard requirement. Do you want to add a suggestion?

Reply:
It’s not a hard requirement. Do you want to add a suggestion?

Reply:
We rely on this signature in the same way as on signatures of PDO, `mysqli` and other extensions, all of which are optional.

Looks like we have different expectations from PSR-3 in DBAL. As I see it, it will just enable dumping debug information into an abstract channel as requested in https://github.com/doctrine/dbal/pull/2911. We're not deprecating `SQLLogger` which has a more specific interface oriented on query execution, not just messages. Therefore, no need to introduce a hard dependency on PSR-3.